class/practice5.1/include/MegaData.hpp

```diff
@@ -25,17 +27,25 @@ class MegaData {
 
 class MegaDataPool {
 public:
-    MegaDataPool(int size) : pool(size), availableObjects(size, true) {}
+
+    static MegaDataPool& getInstance(int size=5) {
```

Comment:
[[nodiscard]] is definitely reasonable here. In general for other getters too but forgettiing about int is not so critical. but if someone asks for such a big object as MegaDataPool and ignores it then something wrong with that code